.evergreen/abi-stability-setup.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+: "${cxx_standard:?}" # Set by abi-stability-checks-* build variant definition.
+
+command -V git >/dev/null
+
+# Files prepared by EVG config.
+[[ -d "mongoc" ]] || {
+  echo "missing mongoc" 1>&2
+  exit 1
+}
+[[ -d "mongo-cxx-driver" ]] || {
+  echo "missing mongo-cxx-driver" 1>&2
+  exit 1
+}
+
+declare working_dir
+working_dir="$(pwd)"
+
+declare cmake_binary
+# shellcheck source=/dev/null
+. ./mongoc/.evergreen/scripts/find-cmake-latest.sh
+cmake_binary="$(find_cmake_latest)"
+command -V "${cmake_binary:?}"
+
+# To use a different base commit, replace `--abbrev 0` with the intended commit.
+# Note: EVG treat all changes relative to the EVG base commit as staged changes!
+declare base current
+base="$(git -C mongo-cxx-driver describe --tags --abbrev=0)"
+current="$(git -C mongo-cxx-driver describe --tags)"
+
+echo "Old Version (Base): ${base:?}"
+echo "New Version (Current): ${current:?}"
+
+printf "%s" "${base:?}" >base-commit.txt
+printf "%s" "${current:?}" >current-commit.txt
+
+# Remove 'r' prefix in version string.
+declare old_ver new_ver
+old_ver="${base:1}"
+new_ver="${current:1}"
+
+declare parallel_level
+parallel_level="$(("$(nproc)" + 1))"
+
+# Use Ninja if available.
+if command -V ninja; then
+  export CMAKE_GENERATOR
+  CMAKE_GENERATOR="Ninja"
+else
+  export CMAKE_BUILD_PARALLEL_LEVEL
+  CMAKE_BUILD_PARALLEL_LEVEL="${parallel_level:?}"
+fi
+
+# Use ccache if available.
+if command -V ccache; then
+  export CMAKE_C_COMPILER_LAUNCHER CMAKE_CXX_COMPILER_LAUNCHER
+  CMAKE_C_COMPILER_LAUNCHER="ccache"
+  CMAKE_CXX_COMPILER_LAUNCHER="ccache"
+fi
+
+# Install prefix to use for ABI compatibility scripts.
+mkdir -p "${working_dir}/install"
+
+# As encouraged by ABI compatibility checkers.
+export CFLAGS CXXFLAGS
+CFLAGS="-g -Og"
+CXXFLAGS="-g -Og"
+
+# Build and install the base commit first.
+git -C mongo-cxx-driver stash push -u
+git -C mongo-cxx-driver reset --hard "${base:?}"
+
+# Install old (base) to install/old.
+echo "Building old libraries..."
+[[ -d install/old ]] || {
```

Comment:
I expect `install/old` is not expected to exist since the teardown task runs `rm -rf *`.

Consider changing to assert: `[[ ! -d install/old ]]`, or removing the `[[ -d install/old ]]` since it suggests there is an expectation of reuse.

Same suggestion applies to `install/new` below.

Reply:
Leftovers from when developing and debugging the setup/teardown task definitions. Will remove.